src/libraries/System.Net.Sockets/src/System/Net/Sockets/SafeSocketHandle.cs

```diff
@@ -43,7 +43,9 @@ public SafeSocketHandle(IntPtr preexistingHandle, bool ownsHandle)
 
         internal bool OwnsHandle { get; }
 
-        internal bool PreferInlineCompletions { get; set; } = true;
+        private static readonly bool PreferInlineCompletionsDefault = Environment.GetEnvironmentVariable("DOTNET_SYSTEM_NET_SOCKETS_INLINE_COMPLETIONS_DEFAULT") == "1";
```

Comment:
This is a temporary envvar for tests and experiments.

Reply:
How does this differ from SocketAsyncEngine.InlineSocketCompletionsEnabled ?

Reply:
> How does this differ from SocketAsyncEngine.InlineSocketCompletionsEnabled ?

`PreferInlineCompletions` defaults to `false` to continue dispatching to `ThreadPool` for `Sockets` which haven't explicitly set the property.

I've added the envvar for the test. It may be also interesting for doing some experiments, like Inline Sockets for database access without having to change the db provider code.

If `PreferInlineCompletions` gets productized/removed, we'll remove this envvar.

Reply:
I've added a comment.